dev-python/networkx/networkx-2.2.recipe

```diff
@@ -0,0 +1,83 @@
+SUMMARY="Python package for handling complex networks"
+DESCRIPTION="NetworkX is a Python package for the creation, manipulation, and \
+study of the structure, dynamics, and functions of complex networks."
+HOMEPAGE="https://networkx.github.io/"
+COPYRIGHT="2004-2018 NetworkX Developers"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/networkx/networkx/archive/networkx-$portVersion.tar.gz"
+CHECKSUM_SHA256="74efce06507cdc2e506c6b7d485a18617adc3a7f355e2dc48ca71c83929bc679"
+SOURCE_DIR="networkx-networkx-$portVersion"
+
+ARCHITECTURES="any"
+
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	setuptools_python
+	setuptools_python3
+	"
+
+BUILD_PREREQUIRES="
+	cmd:python
+	cmd:python3
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+	pythonPackage=${PYTHON_PACKAGES[i]}
+	pythonVersion=${PYTHON_VERSIONS[$i]}
+
+	eval "PROVIDES_$pythonPackage=\"
+		${portName}_$pythonPackage = $portVersion
+		\""
+	eval "REQUIRES_$pythonPackage=\"
+		haiku
+		cmd:python$pythonVersion
+		\""
+done
+
+BUILD()
+{
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
+
+		rm -rf "$sourceDir"-$pythonPackage
+		cp -a "$sourceDir" "$sourceDir"-$pythonPackage
+		cd "$sourceDir"-$pythonPackage
+
+		python=python$pythonVersion
+		$python setup.py build
+	done
+}
+
+INSTALL()
+{
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
+
+		python=python$pythonVersion
+		installLocation=$prefix/lib/$python/vendor-packages/
+		export PYTHONPATH=$installLocation
+		mkdir -p "$installLocation"
+		$python setup.py install \
+			--root=/ --prefix="$prefix"
+
+		packageEntries $pythonPackage \
+			"$prefix"/lib/$python
+	done
+
+	mkdir -p "$developDocDir"
+	cp -r "$prefix"/share/doc/*/./examples "$developDocDir"
+	cp -r "$prefix"/share/doc/*/./requirements.txt "$developDocDir"
+	rm -r "$prefix"/share
```

Comment:
Suggestion:
```diff
 	mkdir -p "$developDocDir"
-	cp -r "$prefix"/share/doc/*/./examples "$developDocDir"
+	mv "$prefix"/share/doc/*/examples "$developDocDir"
-	cp -r "$prefix"/share/doc/*/./requirements.txt "$developDocDir"
-	rm -r "$prefix"/share
+	rm -rf "$prefix"/share
```
`requirements.txt` is not very useful. It only has these 2 lines:
```
-r requirements/default.txt
-r requirements/test.txt
```
but the 2 files which are mentioned are not being installed, so maybe we should just skip that file, `requirements.txt`.
That said, you might wish to keep that file and install the `requirements` subdirectory.
(I don't really know what is best.)

P.S. The `rm -rf` is slightly safer because `INSTALL` would fail if write-protected files get installed by `setup.py` in future relases.